### PR TITLE
Add base management command for projects

### PR DESCRIPTION
**Pull request by c0rydoras**

### CODE REVIEW

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Nice work :+1:

Regarding commit names, it would be nice if they would more reflect a reasoning. For example the commit `refacor(api): drop async in synchronizer`: When reading the code change it's clear what was done. What's unclear is why was it done.

File: api/outdated/commands.py [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+
+            if nonexistant_projects:
+                self.stderr.write(
+                    f"Projects with names {nonexistant_projects} do not exist",
+                )
+                return
+        projects = (
+            Project.objects.filter(id__in=[project.pk for project in projects])
+            or Project.objects.all()
+        )
```
[the remark this answers is not in the document]
Comment (c0rydoras) [reply api/outdated/commands.py]: Sadly using `iexact` like this is not supported :(

but something like this works:

```py
    def handle(self, *args, **options):
        if not options["all"]:
            project_names = options["projects"]
            q = Q()
            for name in project_names:
                q = q | Q(name__iexact=name)
            projects = Project.objects.filter(q)

            if projects.count() != len(project_names):
                missing_projects = set(project_names).difference(
                    projects.values_list("name", flat=True),
                )
                self.stderr.write(
                    f"Projects with names {', '.join(missing_projects)} do not exist",
                )
                return
        else:
            projects = Project.objects.all()

        for project in projects:
            self._handle(project)
```

changes implemented here: https://github.com/adfinis/Outdated/pull/73/commits/67c7afd2d62e6cf2b5f9d4c583f7120fd17c30f0

File: api/outdated/tests/test_commands.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+@pytest.mark.parametrize("all", [True, False])
+def test_command_handle(transactional_db, project_factory, all, mocker):
+    projects = project_factory.create_batch(5)
+    argv = ["", "project-command-test"]
+    handle_mocker = mocker.patch.object(ProjectCommand, "_handle")
+    ProjectCommand().run_from_argv(
+        [*argv, *(["--all"] if all else [project.name for project in projects])]
+    )
+    handle_mocker.assert_has_calls([call(project) for project in projects], True)
```
[the remark this answers is not in the document]
Comment (c0rydoras) [reply api/outdated/tests/test_commands.py]: implemented here: https://github.com/adfinis/Outdated/pull/73/commits/fc22aac26c96e6c7eb6ba2e5f282efe6266608c7

File: api/outdated/commands.py [hunk 3]
```diff
@@ -1,10 +1,40 @@
...
+        projects.add_argument("projects", nargs="*", type=str, default=[])
 
     def handle(self, *args, **options):
-        run(self._handle(*args, **options))
+        if not options["all"]:
+            project_names = options["projects"]
+            q = Q()
+            for name in project_names:
+                q = q | Q(name__iexact=name)
```
Comment (open-dynaMIX) [inline api/outdated/commands.py:22]:
```suggestion
            q = reduce(operator.or_, (Q(name__iexact=name) for name in project_names))
```

Also needs

```python
import operator
from functools import reduce
```

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]

File: api/outdated/commands.py (hunk 3 above)
Comment (c0rydoras) [reply api/outdated/commands.py]: implemented here: https://github.com/adfinis/Outdated/pull/73/commits/89c10b5c2eb2f1035c2d7a6278d23edeee50efab

**Review by open-dynaMIX — APPROVED** [review]

**Pull request merged**